### ISSUE TITLE
Use semver for releases

### ISSUE THREAD

**Issue by gajus** [vercel/turbo#5983]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

At the moment, releases are extremely hard to follow because they batch together helluva lot changes that are not categorized.

Example, if we take a look at [v1.10.14 release notes](https://github.com/vercel/turbo/releases) it lists 95 changes. Some of them have categorization (prefixes such as `fix:`, `test:`, `feat:`), others do not.

I would love to see turbo adopt [`semantic-release`](https://github.com/semantic-release/semantic-release) – make a release every time changes are merged to `main`, require that each PR is merged with a proper category tag, and use `semantic-release` to adjust version depending on the contents of the merged changes.

The primary benefit of the proposed workflow is that it produces smaller and semantically correct releases. Smaller releases mean that it is easier to know what you are getting with every update. Also, if an issue is introduced in one of the releases, then reverting those changes as well as downgrading if necessary, becomes a lot simpler.

### Describe the solution you'd like

N/A

### Describe alternatives you've considered

N/A

<sub>[TURBO-1390](https://linear.app/vercel/issue/TURBO-1390/use-semver-for-releases)</sub>

**Comment by arlyon** [comment]
Hey gajus!

We are currently in the final stages of our go-to-rust port which makes it non-trivial to push new releases on every minor change. We run canary releases which are dogfooded internally for any larger changes, but I am open to shortening the release cadence when we finish the rust port and lower the 'turbulence'. I am bringing this up with our team today to discuss.

Cheers

Alex

**Comment by anthonyshew** [comment]
Our release process is now documented here: https://turbo.build/repo/docs/support#releases (has been for a minute but forgot to update here)
